### ISSUE TITLE
Update GREEND converter

### ISSUE THREAD

**Issue by PMeira** [nilmtk/nilmtk#680]
Currently the GREEND converter seems to have multiple failures. Some are certainly related to changes in Pandas and should be easier to fix.

Related issues (some outdated, the dataset had an update):
- ~~#156~~
- ~~#356~~
- ~~#443~~
- ~~#619~~
 
(Note that #169 will be needed so functions that expect aggregate meters work properly.)

**Comment by PMeira** [comment]
This week I'll try to work on this since there are several related issues.

**Comment by PMeira** [comment]
Besides the usual Pandas updates, the GREEND files need some special treatment. I added this in the function that processes the files:

> GREEND files can be interpreted as multiple CSV blocks concatenated into
> a single file per date. Since the columns of the individual blocks can 
> vary in a single file, they need to be read separately.
> 
> There are some issues we need to handle in the converter:
> - the headers from the multiple blocks
> - corrupted data (lines with null chars, broken lines)
> - more fields than specified in header

Since I had to run the converter multiple times to test and fix issues with the files, I also added an option to use multiprocessing, and lowered the compression level (zlib with level 9 only yielded something like 0.3% smaller size but a very long compression time).  It runs 3 to 4x faster with these two changes.

There could be a issue with the timestamps of some files but unfortunately there's not enough documentation about that, so I'll handle it if there's any report in the future (the original code didn't have any special treatment for this timestamp issue either).

I still need to check the metadata issues (#156 and #619) before closing this.
